lib/evaluate_board.rb

```diff
@@ -16,14 +16,22 @@ def initialize(board_gateway)
   def execute()
     return 'Player one wins' if win_for_player?('X')
     return 'Player two wins' if win_for_player?('O')
-    board = @board_gateway.fetch_board
-    return 'Game over' if board.compact.empty?
+    return 'Game over' if game_over?
 
     ''
   end
 
   private
 
+  def game_over?
+    board = @board_gateway.fetch_board
+    if board.nil?
+      nil
+    else
+      return true if board.compact.length == 9
+    end
+  end
+
   def win_for_player?(player)
```

Comment:
I think instead of this code snippet within `#win_for_player?`:
```ruby
@winning_combinations.each do |combination|
  return true if combination.all? do |num|
    players_marks.include?(num)
  end
end
false
```

You can just do `@winning_combinations.include?(player_marks)`. I've tested it and it still passes all your current tests. I found this more readable?

Reply:
```ruby
def win_for_player?(player)
  players_marks = []
  board = @board_gateway.fetch_board
  if board.nil?
    nil
  else
    board.each_with_index { |cell, i| players_marks << i if cell == player }
    @winning_combinations.include?(players_marks)
  end
end
```

```ruby
def win_for_player?(player)
  board = @board_gateway.fetch_board
  return nil if board.nil?

  players_marks = []
  board.each_with_index { |cell, i| players_marks << i if cell == player }
  @winning_combinations.include?(players_marks)
end
```

Reply:
Does not work for a later test, unfortunately.

Reply:
This solution is failing if the player has more marks than winning_combinations.length